src/libstore/derivations.hh

```diff
@@ -87,10 +88,38 @@ Derivation readDerivation(const Store & store, const Path & drvPath);
 // FIXME: remove
 bool isDerivation(const string & fileName);
 
-Hash hashDerivationModulo(Store & store, const Derivation & drv, bool maskOutputs);
+typedef std::variant<
+    Hash, // regular DRV normalized hash
+    std::map<std::string, Hash> // known CA drv's output hashes
+> DrvHashModulo;
+
+/* Returns hashes with the details of fixed-output subderivations
+   expunged.
+
+   A fixed-output derivation is a derivation whose outputs have a
+   specified content hash and hash algorithm. (Currently they must have
+   exactly one output (`out'), which is specified using the `outputHash'
+   and `outputHashAlgo' attributes, but the algorithm doesn't assume
+   this). We don't want changes to such derivations to propagate upwards
```

Comment:
Since this whole sentence is in parentheses, it makes sense to keep the punctuation inside the parentheses as well.

```suggestion
   this.) We don't want changes to such derivations to propagate upwards
```

Reply:
Indeed it does. I like to violate the rules and do both `".` and `."` for the same reasons. Nesting matters!

Reply:
Honestly, the rule for `".` and `."` is super strange. When quoting, I don't want punctuation that wasn't originally there to be inside the quote... but leaving it outside the quote just looks strange. It's why I try to pad it with extra words before the end of a sentence. English is weird.